src/Widgets/NavigationWidget/SubComponents/LanguageSwitch.tsx

```diff
@@ -0,0 +1,89 @@
+import { NavDropdown } from 'react-bootstrap'
+import {
+  ImageTag,
+  InPlaceEditingOff,
+  LinkTag,
+  Obj,
+  connect,
+  currentPage,
+  currentSiteId,
+} from 'scrivito'
+import { HomepageInstance } from '../../../Objs/Homepage/HomepageObjClass'
+
+export const LanguageSwitch = connect(function LanguageSwitch() {
+  const versions = Obj.onAllSites()
+    .where('_path', 'equals', '/')
+    .toArray()
+    .map((site) => {
+      const siteId = site.siteId()
+      const pageVersion = siteId && currentPage()?.versionOnSite(siteId)
+      const language = site.language() || 'en'
+      return {
+        label: displayName(language),
+        language,
+        version: pageVersion,
+        root: site as HomepageInstance,
+      }
+    })
+    .sort((a, b) => a.label.localeCompare(b.label))
```

Comment:
`localeCompare` also allows to set a `locales` as a second parameter. Not sure, if we should use the current language of the site for that. What do you think?